### DISCUSSION TITLE
Suggestions for new plugins to port

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [madskjeldgaard/portedplugins#23] [Ideas]
Hello everyone

Post your ideas in this thread if you've seen/heard algorithms somewhere that you think would be cool to have in SuperCollider. The only requirements for candidate code is that it has to be open source of course and written in C / C++. Potential candiates can be VCV rack stuff, PD, CSound, VST plugins etc. As long at is written in C or, preferably, C++.

Here are some ideas that I have on the table for now:

- [Surge](https://github.com/surge-synthesizer/surge):  BBDEnsemble, GraphicEQ11Band, ThreelerFilter
- [ComplexNonlinearities](https://github.com/jatinchowdhury18/ComplexNonlinearities) - some of this has been ported already but would be nice to have everything
- [Odin2](https://github.com/TheWaveWarden/odin2) - ParametricEQ

**Comment by madskjeldgaard** [comment]
I think we especially need cool, modern dynamics stuff. Has anyone seen anything cool open source stuff in this realm? Limiters, compressors, etc

**Reply by spluta** [reply]
Most definitely these. Poked around a bit and found this thread:

[https://forum.juce.com/t/is-there-an-open-source-compressor/26721/7](url)

I played around with the IEM and that is quite nice. There is also an ATK Side-Chain Compressor with open source.

**Reply by madskjeldgaard** [reply]
Cool suggestions. Thanks !

**Comment by devilfish707** [comment]
a bunch of zdf filters like these (the ssm, Curtis, ms20 etc are all nice): https://forum.cockos.com/showthread.php?t=213269

things like vladg molot,  an SSL 4K buscomp clone would be cool... but not open source :/ 
(https://www.patreon.com/analogobsession some of this stuff rocks but not open source either)

**Comment by danielmkarlsson** [comment]
This here is my absolute all time fav. There really is nothing else that sounds close to what this sounds like:
https://github.com/dozius/DtBlkFx

A close second are the DestroyFX plugins. My fav is Geometer as that is to my ears the most unique sounding:
https://github.com/sophiapoirier/destroyfx

Another dream of mine would be to have Soundhack Bubbler as a Ugen. A while back it got ported to PD and Max and the source code for those are available here:
http://www.soundhack.com/sh-content/uploads/2011/02/soundhack-0.0.8-src.zip

I've made a lot of music with these tools in case you want to get a feel for what they sound like before you decide whether or not to devote time to porting any of this stuff. You can listen to a bunch of it here:
https://danielmkarlsson.com/ruins-in-the-distance/

**Comment by cianoc** [comment]
Jatin Chowdbury has been busy:
https://github.com/jatinchowdhury18/KlonCentaur
https://github.com/Chowdhury-DSP/ChowKick

Also the https://www.airwindows.com/ stuff is being gradually opensourced.  Though just sifting through his plugins is a project in itself.

**Reply by madskjeldgaard** [reply]
thanks for the suggestions. I think some of the klon centaur stuff is already ported in the analogvintage distortion we have. But will have to double check. The kick  - yes! Let's do it! 

I've also been eyeying the airwindows stuff. If you try anything cool that you think would be nice to have let me know. Thnks

**Comment by madskjeldgaard** [comment]
There's more tape things in the chow dsp vcv project now... https://github.com/jatinchowdhury18/ChowDSP-VCV
